linkerd/app/inbound/src/port_policies.rs

```diff
@@ -36,36 +43,34 @@ pub enum AllowPolicy {
 #[derive(Default)]
 struct PortHasher(u16);
 
-type Map = HashMap<u16, AllowPolicy, BuildHasherDefault<PortHasher>>;
+type Map = HashMap<u16, Arc<ServerPolicy>, BuildHasherDefault<PortHasher>>;
 
 #[derive(Clone, Debug, Error)]
 #[error("connection denied on unknown port {0}")]
-pub struct DeniedUnknownPort(u16);
+pub(crate) struct DeniedUnknownPort(u16);
 
-#[derive(Clone, Debug, Error, PartialEq, Eq)]
-#[error("expected one of `deny`, `authenticated`, `unauthenticated`, or `tls-unauthenticated`")]
-pub struct ParsePolicyError(());
+#[derive(Debug, thiserror::Error)]
```

Comment:
nit: `Error` is already imported
```suggestion
#[derive(Debug, Error)]
```